providers/providerverifier.go

```diff
@@ -137,39 +138,39 @@ func (v *DefaultProviderVerifier) VerifyIDToken(ctx context.Context, idToken []b
 
 	switch alg {
 	case gq.GQ256:
+		// GQ signatures need special handling (extract original headers, etc.)
 		if err := v.verifyGQSig(ctx, idt); err != nil {
 			return fmt.Errorf("error verifying OP GQ signature on PK Token: %w", err)
 		}
-	case jwa.RS256:
-		pubKeyRecord, err := v.providerPublicKey(ctx, idToken)
-		if err != nil {
-			return fmt.Errorf("failed to get OP public key: %w", err)
-		}
-
-		// Ensure that the algorithm of public key from OpenID Provider matches the algorithm specified in the ID Token
-		if _, ok := pubKeyRecord.PublicKey.(*rsa.PublicKey); !ok {
-			return fmt.Errorf("public key is not an RSA public key")
-		}
-
-		if _, err := jws.Verify(idToken, jws.WithKey(alg, pubKeyRecord.PublicKey)); err != nil {
-			return err
-		}
-	case jwa.ES256:
+	default:
+		// Generic verification for RS256, ES256, EdDSA, and future algorithms
 		pubKeyRecord, err := v.providerPublicKey(ctx, idToken)
 		if err != nil {
 			return fmt.Errorf("failed to get OP public key: %w", err)
 		}
 
-		// Ensure that the algorithm of public key from OpenID Provider matches the algorithm specified in the ID Token
-		if _, ok := pubKeyRecord.PublicKey.(*ecdsa.PublicKey); !ok {
-			return fmt.Errorf("public key is not an ECDSA public key")
+		// Validate that key type matches the algorithm in the token
+		switch alg {
```

Comment:
This switch inside of a switch weirded me out at first, but I see what your doing here and it makes sense to me.